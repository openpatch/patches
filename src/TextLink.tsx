import { css } from "@emotion/react";
import { forwardRef, ReactNode } from "react";
import { useLinkComponent } from "./hooks";
import { Icon } from "./Icon";
import { ExternalWindow } from "./icons/shade";
import { LinkComponentProps } from "./LinkComponentProvider";

export type TextLinkProps = {
  children?: ReactNode;
} & LinkComponentProps;

export const TextLink = forwardRef<HTMLAnchorElement, TextLinkProps>(
  ({ children, href, color, ...props }, ref) => {
    const LinkComponent = useLinkComponent(ref);
    const external = href[0] !== "/" && href[0] !== "#";
    return (
      <LinkComponent
        css={(theme) => css`
          display: inline;
          color: inherit;
          svg {
            margin-left: ${theme.space.xxsmall};
            width: auto;
            height: 90%;
          }
        `}
        href={href}
        ref={ref}
        {...props}
      >
        {children}
        {external && (
          <Icon size="auto" color="currentColor">
            <ExternalWindow />
          </Icon>
        )}
      </LinkComponent>
    );
  }
);
