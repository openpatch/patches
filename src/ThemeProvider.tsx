import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
