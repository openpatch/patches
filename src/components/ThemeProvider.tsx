import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
