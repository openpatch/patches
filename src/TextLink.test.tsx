import { render } from "@testing-library/react";
import { TextLink, TextLinkProps } from "./TextLink";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: TextLinkProps) =>
  render(
    <ThemeProvider>
      <TextLink href="/" {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
