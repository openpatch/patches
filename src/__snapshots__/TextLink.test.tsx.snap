// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  display: inline;
  color: inherit;
}

.emotion-0 svg {
  margin-left: 4px;
  width: auto;
  height: 70%;
}

<div>
  <div
    id="patches-root"
  >
    <a
      class="emotion-0"
      href="/"
    />
  </div>
</div>
`;
